string direction
float64 cost